pkg/controllers/raycluster_controller.go

```diff
@@ -544,6 +550,51 @@ func desiredHeadNetworkPolicy(cluster *rayv1.RayCluster, cfg *config.KubeRayConf
 		)
 }
 
+func (r *RayClusterReconciler) checkAndDeleteHeadPod(ctx context.Context, cluster *rayv1.RayCluster) error {
+	serviceAccount, err := r.kubeClient.CoreV1().ServiceAccounts(cluster.Namespace).Get(ctx, oauthServiceAccountNameFromCluster(cluster), metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get OAuth ServiceAccount: %w", err)
+	}
+
+	headPod, err := getHeadPod(ctx, r, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get head pod: %w", err)
+	}
+
+	if serviceAccount.ImagePullSecrets == nil || len(serviceAccount.ImagePullSecrets) == 0 {
+		return fmt.Errorf("ServiceAccount missing ImagePullSecrets")
+	}
```

Comment:
Is this really an error?

Reply:
It is not an error, I though it needs to reconcile if serviceAccount has no ImagePullSecrets